### PR TITLE
feature enhancement: add export-package to ask-init to download skill-packages from smapi

### PR DESCRIPTION
**Pull request by CamdenFoucht**
*Description of changes:*

Adding a simple `export-package` option to the `ask init` command.

I have recently been working with a team creating widgets for alexa skills and they are heavy users of `ask init`. But after they use `ask init` and commit changes, they also update the skill via the developer console.

This causes an issue because those changes aren't stored in `git`. So I've added an `export-package` option that will update `ask init` to download files from smapi alongside cloning the git repo.

Before this option, the `ask init` command would only download skill-package assets from smapi if a `skill-package` folder didn't exist in git. So if a developer does commit the skill-package folder, it will no longer use `smapi export-package`. Now when `ask init --hosted-skill-id <id> --export-package` is used, it will always download the skill schema from smapi so developers can recommit these files if wanted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/commands/init/index.js [hunk 1]
```diff
@@ -85,6 +86,9 @@ function initAlexaHostedSkill(rootPath, cmd, profile, cb) {
                             return cb(hooksErr);
                         }
                         Messenger.getInstance().info(`\n${skillName} successfully initialized.\n`);
+                        Messenger.getInstance().warn('If updating the skill package via the developer console, your git repo will be out of sync. '
```
Comment (CamdenFoucht) [inline lib/commands/init/index.js:89]: Any suggestions for a message here?

Basically people need to know if they create a hosted cli project, it doesn't stay synced with the UI dev console changes if they're making changes in both places

File: docs/concepts/Alexa-Hosted-Skill-Commands.md [hunk 2]
```diff
@@ -40,10 +40,11 @@ The user is not prompted for a `skill template`. The [Hello World Skill](https:/
 
 This command initializes Alexa Hosted Skills by cloning the project from the hosted skill service, downloading the latest Alexa skill package, and provide a git-ready environment. Developers can then checkout, pull from, and push to a remote Git repository from their local machines.
 
+If you make changes to the skill package such as the interaction model or publishing information on the Alexa Developer Console, these changes will not be reflected in your git repository. You will have to resync the skill-package which you could you do by running `ask init --hosted-skill-id <id> --export-package` and committing the changes that were made on the console.
```
Comment (doiron) [inline docs/concepts/Alexa-Hosted-Skill-Commands.md:43]: nit: "which you could you do by running"
suggest:  "which you could do by running"

File: lib/commands/init/index.js [hunk 3]
```diff
@@ -85,6 +86,9 @@ function initAlexaHostedSkill(rootPath, cmd, profile, cb) {
...
+                            + 'You can use the "export-package" option with "ask init" to download the skill package and commit the changes to get back in sync.\n');
```
Comment (doiron) [inline lib/commands/init/index.js:90]: nit: "and commit the changes to get back in sync."  is a little bit vague.

File: docs/concepts/Alexa-Hosted-Skill-Commands.md (hunk 2 above)
Comment (CamdenFoucht) [reply docs/concepts/Alexa-Hosted-Skill-Commands.md:43]: Thanks for typo catch :)

File: lib/commands/init/index.js (hunk 3 above)
Comment (CamdenFoucht) [reply lib/commands/init/index.js:90]: Yeah I have a comment above asking for a good suggestion of what we should say here :D Didn't want to write a long paragraph for it

File: docs/concepts/Alexa-Hosted-Skill-Commands.md (hunk 2 above)
Comment (tydonelson) [inline docs/concepts/Alexa-Hosted-Skill-Commands.md:43]: This seems conceptually more like a "sync-package" operation to me.. When I think of "import" I think of the user specifying a location to grab the skill-package from, and "export" would be like dumping the package to a location the user specifies. Am I understanding right that this is just synchronizing the state between the skill stored in the developer console and the local skill package? Just looking for thoughts here around naming :)

File: lib/commands/init/index.js (hunk 1 above)
Comment (doiron) [reply lib/commands/init/index.js:89]: you might want to sync up with a doc writer?

File: docs/concepts/Alexa-Hosted-Skill-Commands.md (hunk 2 above)
Comment (CamdenFoucht) [reply docs/concepts/Alexa-Hosted-Skill-Commands.md:43]: Yeah I agree, I used `export-package` since there's a command

```
ask smapi export-package
```

which downloads it from smapi. (I guess name was chosen since it's exporting it from UI)

`sync-package` is good for more clearness, although it current approach it doesn't necessarily `sync` them since it doesn't commit the changes

File: lib/commands/init/index.js (hunk 3 above)
Comment (tydonelson) [reply lib/commands/init/index.js:90]: Maybe:

"Any changes made to the skill package in the developer console will not be reflected in your local skill package git repository. Use "ask init" with the "export-package" option to import any outstanding changes from the developer console. You can then use git to commit those changes to your skill package repository."

Still a bit wordy, but a little more explicit around what "sync" means

Comment (CamdenFoucht) [reply lib/commands/init/index.js:90]: Perfect 👍

File: docs/concepts/Alexa-Hosted-Skill-Commands.md (hunk 2 above)
Comment (tydonelson) [reply docs/concepts/Alexa-Hosted-Skill-Commands.md:43]: Yeah that's fair. They'll at least be listed as modified files so they'll (likely) get committed at some point. Maybe a point to expand on this in the future, we could stash any local changes, sync with the smapi 'export-package', commit, then pop the stash to re-apply any outstanding local changes?
